lib/readline.js

```diff
@@ -952,11 +993,16 @@ Interface.prototype._ttyWrite = function(s, key) {
         break;
 
       case 'left':
-        this._moveCursor(-1);
+        if (this.cursor > 0) {
+          // obtain the code point to the left
+          this._moveCursor(-codePointLeftOf(this.cursor, this.line).length);
+        }
```

Comment:
Hitting `left` or `right` should always still trigger `_moveCursor`. That way the line is refreshed.